10_Examples/jett-onboarding/CMakeLists.txt

```diff
@@ -0,0 +1,72 @@
+cmake_minimum_required(VERSION 3.8)
+project(jett-onboarding)
+
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic)
+endif()
+
+# find dependencies
+find_package(ament_cmake REQUIRED)
+find_package(rclcpp REQUIRED)
+find_package(std_msgs REQUIRED)
+
+find_package(ament_cmake_gtest REQUIRED)
+
+# uncomment the following section in order to fill in
+# further dependencies manually.
+# find_package(<dependency> REQUIRED)
+add_executable(listener src/subscriber_member_function.cpp)
+ament_target_dependencies(listener rclcpp std_msgs)
+
+add_executable(talker src/publisher_member_function.cpp)
+ament_target_dependencies(talker rclcpp std_msgs)
+install(TARGETS
+  talker
+  listener
+  DESTINATION lib/${PROJECT_NAME})
+
+add_executable(hello_world src/hello_world.cpp)
+install(TARGETS
+  hello_world
+  DESTINATION lib/${PROJECT_NAME})
+target_link_libraries(hello_world
+talking
+)
+
+# Add Class as Library to allow it to share across files easier probably
+add_library(talking SHARED src/talking.cpp)
```

Comment:
Fun fact time:

There are mainly two types of libraries: Dynamic/Shared and Static. 

When you link a static library to another target (lib or exe), it basically copy pasts all the cpp files into that same executable.

When you link a shared library, the other program is gonna look for 1's and 0's for that library with you actually run the program. So, if we built everything, and then someone went in and deleted my_shared_library.so, then any exes or libs that depended on it would break when you ran them.

Not gonna go into why you would choose one or the other, but good to have heard of the concept before.